### ISSUE TITLE
Ref reassignment should propagate `readonly` as proposed.

### ISSUE THREAD

**Issue by VSadov** [dotnet/roslyn#24874]
- if LHS is readonly, the whole expression is readonly

It looks like the logic above is NYI.

Example:

```C#
using System;
public class C {
    
    public readonly int ro = 42;
    
    public int rw = 111;
    
    public void M() 
    {
        // just init to something
        ref readonly var roR = ref ro;
        ref var rwR = ref rw;
       
        // tests
        rwR = ref (roR = ref rwR);       // should this be an error? 
        
        
        // looks like we might have bugs here
        roR = ref (roR = ref ro);        // should not be an error
                
        rwR = ref (roR = ref ro);        // should be an error
    }
}
```

**Comment by VSadov** [comment]
CC: @agocke

**Comment by agocke** [comment]
Fixed by #24904
